### PR TITLE
Subtree push with 1.77.0 nightly 2023-12-28

### PR DESCRIPTION
**Pull request by ytmimi**
I figured I'd get another subtree push done in an effort to do them on a more regular basis. 

r? @calebcartwright

Here's the link to the [Diff-Check job](https://github.com/rust-lang/rustfmt/actions/runs/7352666378)

### CODE REVIEW

File: src/types.rs
Comment (ytmimi) [inline src/types.rs:1]: All of these changes are related to https://github.com/rust-lang/rust/pull/119163

**Review by compiler-errors — APPROVED** [review]
I reviewed every PR that went into this sync, so let me know if there are any questions. LGTM though.

**Comment by ytmimi** [comment]
@compiler-errors Thanks! I'm still not sure when we'll do the full subtree-sync with `r-l/rust`, but in an effort to make that process a little easier we're trying to do more regular subtree pushes.

**Review by calebcartwright — APPROVED** [review]
thank you both

**Pull request merged**